### PR TITLE
Support annotation context / classids for `GeoPoints`

### PR DESCRIPTION
**Pull request by Wumpf**
### What

In order to do so...
* moves annotation scene context system to re_space_view
* move some utilities for resolving annotation contexts to re_space_view
    * doing so, I noticed that it always processes keypoints, so I added a variant without keypoints to it
    * doing so I noticed that _almost everything_ queries keypoints for no reason at all - keypoints are only supported on points 2d & 3d (check the fbs definitions! also the way we speced it it doesn't make sense otherwise)
         * -> rip out a lot of of keypoint queries 
* move color resolve utility to re_space_view
* breathe

This allows me to have automagically colored points on nuscenes! (separate pr for this update coming soon!)
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/8124?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/8124?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/8124)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

To deploy documentation changes immediately after merging this PR, add the `deploy docs` label.

### CODE REVIEW

File: crates/viewer/re_space_view/src/annotation_context_utils.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                class_description.annotation_info_with_keypoint(keypoint_id.0)
+            })
+            .collect();
+
+        (ResolvedAnnotationInfos::Many(annotation_info), keypoints)
+    }
+}
+
+/// Resolves all annotations for the given entity view.
+pub fn process_annotation_slices(
```
Comment (Wumpf) [inline crates/viewer/re_space_view/src/annotation_context_utils.rs:121]: this method is new. Everything else is just moved code, didn't touch it one bit

File: crates/viewer/re_space_view/src/annotation_context_utils.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+use re_types::components::Color;
+use re_viewer_context::{Annotations, QueryContext, ResolvedAnnotationInfos};
+
+use crate::clamped_or_nothing;
+
+#[inline]
+fn to_egui_color(color: &Color) -> egui::Color32 {
+    let [r, g, b, a] = color.to_array();
+    egui::Color32::from_rgba_unmultiplied(r, g, b, a)
+}
```
Comment (emilk) [inline crates/viewer/re_space_view/src/annotation_context_utils.rs:12]: We already have this:
https://github.com/rerun-io/rerun/blob/7841b5ede6ae613781d7adfe6002287e14332f80/crates/store/re_types/src/components/color_ext.rs#L57-L64

File: crates/viewer/re_space_view/src/annotation_context_utils.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+/// Process [`Color`] components using annotations and default colors.
+pub fn process_color_slice<'a>(
+    ctx: &QueryContext<'_>,
+    fallback_provider: &'a dyn re_viewer_context::TypedComponentFallbackProvider<Color>,
+    num_instances: usize,
+    annotation_infos: &'a ResolvedAnnotationInfos,
+    colors: &'a [Color],
+) -> Vec<egui::Color32> {
+    // NOTE: Do not put tracing scopes here, this is called for every entity/timestamp in a frame.
```
Comment (emilk) [inline crates/viewer/re_space_view/src/annotation_context_utils.rs:22]: We could still have `puffin::profile_function_if(10_000 < num_instances);` to catch the cost of e.g. a big point cloud, without having to pay any overhead for small entities

**Review by emilk — APPROVED** [review]
Nice cleanup!

**Pull request merged**